src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+# See: https://spdx.org/licenses/
+
+
+from typing import Optional, Union
+import numpy as np
+import networkx as ntx
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
+
+
+class SchedulingProblem:
+    def __init__(self,
+                 num_agents: int = 3,
+                 num_tasks: int = 3,
+                 sat_cutoff: Union[float, int] = 0.99,
+                 seed: int = 42):
+        """Schedule `num_tasks` tasks among `num_agents` agents such that
+        every agent performs exactly one task and every task gets assigned
+        exactly one agent.
+
+        Parameters
+        ----------
+        num_agents (int) : number of agents available to perform all tasks.
+        Default is arbitrarily chosen as 3.
+
+        num_tasks (int) : number of tasks to be performed. Default is
+        arbitrarily chosen as 3.
+
+        sat_cutoff (float or int) : If float, this is the satisfiability
+        cut-off calculated as the ratio between the number of tasks that get
+        assigned an agent to the total number of tasks. A fraction between 0
+        and 1. If int, this is the target cost for the underlying QUBO
+        solver. Default is 0.99 (i.e., 99% of the total number of tasks get
+        assigned an agent).
+
+        seed (int) : Seed for PRNG used in problem generation.
+        """
+        self._num_agents = num_agents
+        self._agent_ids = range(num_agents)
+        self._agent_attrs = None
+        self._num_tasks = num_tasks
+        self._task_ids = range(num_tasks)
+        self._task_attrs = None
+        self._sat_cutoff = sat_cutoff
+        self.graph = None
+        self.adjacency = None
+        self._random_seed = seed
+
+    @property
+    def num_agents(self):
+        return self._num_agents
+
+    @num_agents.setter
+    def num_agents(self, val: int):
+        self._num_agents = val
+
+    @property
+    def agent_ids(self):
+        return self._agent_ids
+
+    @property
+    def agent_attrs(self):
+        return self._agent_attrs
+
+    @agent_attrs.setter
+    def agent_attrs(self, attr_vec):
+        self._agent_attrs = attr_vec
+
+    @property
+    def num_tasks(self):
+        return self._num_tasks
+
+    @num_tasks.setter
+    def num_tasks(self, val: int):
+        self._num_tasks = val
+
+    @property
+    def task_ids(self):
+        return self._task_ids
+
+    @property
+    def task_attrs(self):
+        return self._task_attrs
+
+    @task_attrs.setter
+    def task_attrs(self, attr_vec):
+        self._task_attrs = attr_vec
+
+    @property
+    def sat_cutoff(self):
+        return self._sat_cutoff
+
+    @sat_cutoff.setter
+    def sat_cutoff(self, val: float):
+        self._sat_cutoff = val
+
+    @property
+    def random_seed(self):
+        return self._random_seed
+
+    @random_seed.setter
+    def random_seed(self, val: int):
+        self._random_seed = val
+
+    def is_node_valid(self, *args):
```

Comment:
Could you add a docstring to denote what this method is useful for? Why does it just return True?

Reply:
Added a detailed docstring.